File: CSharp/66_PlusOne.cs
namespace LeetCodeSolutions {
    public static class PlusOne {
        public static int[] Solution(int[] digits) {
        for (int i = digits.Length - 1; i >= 0; i--) {
            if (digits[i] < 9) {
                digits[i]++;
                return digits; 
            }
            digits[i] = 0;
        }

        int[] one = new int[] { 1 };
        int[] result = one.Concat(digits).ToArray();
        return result;
        }

        public static void RunTests() {
            Console.WriteLine("Running Length Of Plus One Tests...");

            var testCases = new List<(int[], int[])> {
                (new int[] {1, 2, 3}, new int[] {1, 2, 4}),
                (new int[] {1, 2, 3, 4}, new int[] {1, 2, 3, 5}),
                (new int[] {9, 9, 9}, new int[] {1, 0, 0, 0}),
                (new int[] {4, 3, 2, 1}, new int[] {4, 3, 2, 2}),
                (new int[] {9}, new int[] {1, 0}),
            };

            foreach (var (digits, expected) in testCases) {
                try {
                    var result = Solution(digits);
                    Console.WriteLine(
                        $"Input: {string.Join(",", digits)} " +
                        $"Expected: {string.Join(",", expected)} Result: {string.Join(",", result)}"
                    );
                } catch (Exception ex) {
                    Console.WriteLine($"Test Failed: {ex.Message}");
                }
            }
        }
    }
}

File: JavaScript/66_PlusOne.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    for (let i = digits.length - 1; i > -1; i--) {
        if (digits[i] < 9) {
            digits[i] += 1;
            return digits;
        }
        digits[i] = 0;
    }
    digits.unshift(1)
    return digits;
};


const testCases = [
    {
        digits: [1,2,3],
        expected: [1,2,4]
    },
    {
        digits: [1,2,3,4],
        expected: [1,2,3,5]
    },
    {
        digits: [9,9,9],
        expected: [1,0,0,0]
    },
    {
        digits: [4,3,2,1],
        expected: [4,3,2,2]
    },
    {
        digits: [9],
        expected: [1,0]
    },
]

testCases.forEach((testCase, i) => {
    const output = plusOne(testCase.digits);
    if (JSON.stringify(output) === JSON.stringify(testCase.expected)) {
        console.log(`Test Case ${i}: Passed. Expected: ${JSON.stringify(testCase.expected)}, Output: ${JSON.stringify(output)} `)
    } else {
        console.log(`Test Case ${i}: Failed. Expected: ${JSON.stringify(testCase.expected)}, Output: ${JSON.stringify(output)} `)
    }
});